### PR TITLE
Fix HorizontalPodAutoscaler metrics not displaying current values

### LINKED ISSUE #5942: HPA list no getting current usage
**Describe the bug**
Lens HPA list is not showing the current resources usage. It is doubling the resource limit.
e.g.: for a deployment configured with 40% average cpu Lens is showing 40%/40% in the HPA list.

**To Reproduce**
1. Go to 'Config'
2. Click on 'HPA'
3. Check list HPA list (column Metrics)

**Expected behavior**
HPA list should show the "current utilization/metric value" in the Metrics column

**Screenshots**
image

**Environment (please complete the following information):**
- Lens Version: 6.0.0-Latest-20220728.2
- OS: Windows 11
- Installation method: Lens Auto update

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/5942
Screen Shot 2022-07-29 at 2 10 59 PM

### CODE REVIEW

File: src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts [hunk 1]
```diff
@@ -263,3 +209,107 @@ export class HorizontalPodAutoscalerApi extends KubeApi<HorizontalPodAutoscaler>
...
+        ?? targetMetric?.averageValue
+    ),
+  };
+}
+
+function getExternalMetricValue(currentMetric: ExternalMetricStatus | undefined, targetMetric: ExternalMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentValue
+        ?? currentMetric?.currentAverageValue
```
Comment (ixrock) [inline src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts:271]: _nit:_ why such simple expression with 2 lines?

File: src/renderer/components/+config-autoscalers/hpa-details.tsx [hunk 2]
```diff
@@ -63,15 +63,14 @@ class NonInjectedHpaDetails extends React.Component<HpaDetailsProps & Dependenci
...
             : "";
 
           return (
             <>
-              Resource
+              {"Resource "}
               {metricSpec.name}
-              {" "}
-              on Pods
+              {" on Pods"}
```
Comment (ixrock) [inline src/renderer/components/+config-autoscalers/hpa-details.tsx]: Would be better to put a whole text inside template literal, e.g:

```
return `Resource ${metricSpec.name} on Pods ${addition}`;
```

**Review by ixrock — DISMISSED** [review]

File: src/renderer/components/+config-autoscalers/hpa-details.tsx [hunk 3]
```diff
@@ -98,9 +96,7 @@ class NonInjectedHpaDetails extends React.Component<HpaDetailsProps & Dependenci
           return (
             <>
               {metric.external.metricName}
-              {" "}
-              on
-              {" "}
+              {" on "}
               {JSON.stringify(metric.external.metricSelector)}
```
Comment (ixrock) [inline src/renderer/components/+config-autoscalers/hpa-details.tsx]: Same here (merge into template)

File: src/renderer/components/+config-autoscalers/hpa-details.tsx (hunk 2 above)
Comment (Nokel81) [reply src/renderer/components/+config-autoscalers/hpa-details.tsx]: Sure

File: src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts (hunk 1 above)
Comment (Nokel81) [reply src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts:271]: Personal preference. Mostly because each line is a bit long.

**Review by ixrock — APPROVED** [review]

File: src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts [hunk 4]
```diff
@@ -263,3 +209,107 @@ export class HorizontalPodAutoscalerApi extends KubeApi<HorizontalPodAutoscaler>
...
+      targetMetric?.targetValue
+        ?? targetMetric?.targetAverageValue
+    ),
+  };
+}
+
+function getContainerResourceMetricValue(currentMetric: ContainerResourceMetricStatus | undefined, targetMetric: ContainerResourceMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentAverageUtilization
```
Comment (jim-docker) [inline src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts:283]: nit: if `currentMetric.currentAverageUtilization === 0` shouldn't it really return `0%`?

**Review by jim-docker — APPROVED** [review]

File: src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts (hunk 4 above)
Comment (Nokel81) [reply src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts:283]: Yes it should. Will fix in a follow up PR.

**Pull request merged**